haystack/nodes/prompt/prompt_node.py

```diff
@@ -911,11 +957,13 @@ def run(
         if meta and "meta" not in invocation_context.keys():
             invocation_context["meta"] = meta
 
-        results = self(prompt_collector=prompt_collector, **invocation_context)
+        if "prompt_template" not in invocation_context.keys():
+            invocation_context["prompt_template"] = self.get_prompt_template(prompt_template)
 
-        prompt_template = self.get_prompt_template(self.default_prompt_template)
-        output_variable = self.output_variable or prompt_template.output_variable or "results"
+        results = self(prompt_collector=prompt_collector, **invocation_context)
 
+        prompt_template_resolved: PromptTemplate = invocation_context.pop("prompt_template")
```

Comment:
Pop it, to keep the invocation_context clean and serializable. We also need it to infer the output_variable.